### PR TITLE
Fix for the "Back button can't end Unciv (#1513)"

### PR DESCRIPTION
**Pull request by JackRainy**
The fix for the issue: **Back button can't end Unciv** (#1513 )
1) Compilation fix for the 12a98aa4bb2ee8a136a9b385459809e39d0d173a
2) The exit functionality by itself
3) The nice prompt dialog with a bit of localization

### CODE REVIEW

File: core/src/com/unciv/ui/cityscreen/ConstructionsTable.kt [hunk 1]
```diff
@@ -261,7 +261,7 @@ class ConstructionsTable(val cityScreen: CityScreen) : Table(CameraStageBaseScre
                 && UncivGame.Current.worldScreen.isPlayersTurn
                 && !city.isPuppet) {
             val constructionGoldCost = construction.getGoldCost(city.civInfo)
-            purchaseConstructionButton = TextButton("Buy for [$constructionGoldCost] gold".tr(), CameraStageBaseScreen.skin)
+            val purchaseConstructionButton = TextButton("Buy for [$constructionGoldCost] gold".tr(), CameraStageBaseScreen.skin)
```
Comment (r3versi) [inline core/src/com/unciv/ui/cityscreen/ConstructionsTable.kt]: Maybe strip this change out of the PR: what you see is an artifact introduced by the merge. It will be fixed in the dedicated PR #1648.

File: core/src/com/unciv/UncivGame.kt [hunk 2]
```diff
@@ -24,7 +24,8 @@ import kotlin.concurrent.thread
 
 class UncivGame(
         val version: String,
-        private val crashReportSender: CrashReportSender? = null
+        private val crashReportSender: CrashReportSender? = null,
+        exitEventCallback: ()->Unit
```
Comment (r3versi) [inline core/src/com/unciv/UncivGame.kt]: Maybe change this to
`private val exitEvent: ()->Unit = {}`, so you can get rid of line 43 and don't need to specify an empy callback?

File: core/src/com/unciv/ui/cityscreen/ConstructionsTable.kt (hunk 1 above)
Comment (JackRainy) [reply core/src/com/unciv/ui/cityscreen/ConstructionsTable.kt]: Thank you, will remove that.

File: core/src/com/unciv/UncivGame.kt (hunk 2 above)
Comment (JackRainy) [reply core/src/com/unciv/UncivGame.kt]: Actually, it is not always empty. For Android - there is a `finish()` call. For Desktop - empty, since there is no "Back" button. And according to the failing test, for iOS - there must be something totally different.

Comment (JackRainy) [reply core/src/com/unciv/UncivGame.kt]: Ahh... now I got it. You suggest the default value for the parameter. Good idea, will correct that.

Comment (r3versi) [reply core/src/com/unciv/UncivGame.kt]: Of course, if it were always empty there would be no need of a callback I guess. The change I proposed is to let you create a game instance without specifying an `exitEvent`.

File: core/src/com/unciv/ui/cityscreen/ConstructionsTable.kt (hunk 1 above)
Comment (JackRainy) [reply core/src/com/unciv/ui/cityscreen/ConstructionsTable.kt]: I need to keep this change for a while, in order to ensure that the tests are green, not failing due to compilation error. I assume it would be eliminated during rebase when the PR be applied.

File: core/src/com/unciv/UncivGame.kt [hunk 3]
```diff
@@ -24,8 +24,12 @@ import kotlin.concurrent.thread
 
 class UncivGame(
         val version: String,
-        private val crashReportSender: CrashReportSender? = null
+        private val crashReportSender: CrashReportSender? = null,
+        val exitEvent: ()->Unit = {}
```
Comment (naivekook) [inline core/src/com/unciv/UncivGame.kt]: actually more Kotlin idiomatic to make in nullable and null as default
because in current situation we have empty func which will run every time

**Comment by JackRainy** [comment]
New PR #1661 is created.
